[Bug] Fixed geo-filter extra layer issue

macrigiuseppe opened the pull request:
Fix #935 
The geo-filter would not work on layers added after the first one

### Before:

![geo-feature-filter-bug](https://user-images.githubusercontent.com/2554552/73575183-a03ce900-4445-11ea-9042-afeceb29e271.gif)

### After:

![geo-feature-filter-solution](https://user-images.githubusercontent.com/2554552/73575187-a4690680-4445-11ea-8275-26a9d996a8f4.gif)


Signed-off-by: Giuseppe Macri <gmacri@uber.com>

heshan0131 commented:
you might need to rebase so that my changes aren't included in here

macrigiuseppe commented:
> you might need to rebase so that my changes aren't included in here

@heshan0131 rebased with master

heshan0131 commented:
> > you might need to rebase so that my changes aren't included in here
> 
> @heshan0131 rebased with master

I still see previous commit, did you `push -f`?

macrigiuseppe commented:
> > > you might need to rebase so that my changes aren't included in here
> > 
> > 
> > @heshan0131 rebased with master
> 
> I still see previous commit, did you `push -f`?


@heshan0131 yeah, i think i rebased before i merged your changes onto master. 
Rebased again

heshan0131 commented on src/reducers/vis-state-updaters.js:
```diff
@@ -585,23 +587,30 @@ export function setFilterUpdater(state, action) {
+      datasetIds = layerDataIds;
+
+      // Update newFilter dataIds
+      const newDataIds = uniq(
```
simply use `layerDataIds ` seems sufficient, why do you have to recalculate `newDataIds`?

macrigiuseppe replied:
LayerDataIds is used to determine which datasets to perform the filter against.
The use case is the following:
- add a layer A (dataset_1) to the filter:
  - layerDataIds: [dataset_1]
  - newDataIds: [dataset_1]

- add a layer B (dataset_2) to the filter:
  - layerDataIds: [dataset_2]
  - newDataIds: [dataset_1, dataset_2]

- add a layer C (dataset_1) to the filter:
  - layerDataIds: [dataset_1]
  - newDataIds: [dataset_1, dataset_2]

__layerDataIds__ is necessary to determine which dataset to run the filter against during the current step.
__newDataIds__ is required to update the filter object with. the list of unique dataset ids
